stdlib/REPL/src/docview.jl

```diff
@@ -452,13 +452,21 @@ function repl_latex(io::IO, s::String)
 end
 repl_latex(s::String) = repl_latex(stdout, s)
 
+function normalize_symbol(s::Symbol)
+    # parse to apply string->normsymbol
+    normalized = Meta.parse(string(s), raise = false, depwarn = false)
+    normalized isa Symbol ? normalized : s
+end
+
 macro repl(ex, brief::Bool=false) repl(ex; brief=brief) end
 macro repl(io, ex, brief) repl(io, ex; brief=brief) end
 
 function repl(io::IO, s::Symbol; brief::Bool=true)
+    str_orig = string(s)
+    s = normalize_symbol(s)
```

Comment:
Who is constructing symbols explicitly, without having parsed them?

Reply:
That's quite early in `_helpmode(io::IO, line::AbstractString)`. The whole input line is made into a symbol (apart from stripped white space at the ends):
https://github.com/JuliaLang/julia/blob/e66091887b2882e2df7324fd7710a46aa75e0289/stdlib/REPL/src/docview.jl#L48
In the case of `\minus`, `\cdot`, and `\cdotp`, the if branch directly afterwards is executed because `Base.isoperator(assym) == true`, and so `@repl` is called with `assym`, which in turn calls `repl(io::IO, s::Symbol; brief::Bool=true)`.

Reply:
The information which exact characters were entered is needed at that point for `repl_latex($io, $str_orig)`, which shows
```
"−" can be typed by \minus<tab>
```

So, constructing the symbol without parsing seems to be a sort of trick to carry two kinds of information (the exact input and the fact that it will be parsed as a keyword/operator/...) in one macro/function argument on which we can dispatch.

Reply:
I just dug a bit deeper and found #19464. µ (U+00B5 micro) and μ (U+03BC greek small letter mu) both show
```
"μ" can be typed by \mu<tab>
```
which is technically only true for the latter. This is because the code goes through the `else` branch, so the symbol is not `assym` but `x` which comes from `Meta.parse`. But the binding is shown correctly, which is much more important.

In it's current state, this PR is really only a quick fix for specific cases which behave clearly wrong in my opinion. For anything further, the desired behavior should be decided on first.

Reply:
And I found out why the symbols are constructed explicitly without parsing. That was introduced by #35154 because of this:
```julia
julia> x = Meta.parse("+=", raise = false, depwarn = false)
:($(Expr(:error, "invalid identifier name \"+=\"")))

julia> x = Meta.parse(".=", raise = false, depwarn = false)
:($(Expr(:error, "invalid identifier name \".=\"")))
```

Reply:
Something like this could be better for `+=` and `.=`, what do you say?
```julia
x = Meta.parse(line, raise = false, depwarn = false)
if Meta.isexpr(x, :error)
    asinfix = Meta.parse("x "*line*" x", raise = false, depwarn = false)
    if asinfix isa Expr && length(asinfix.args) == 2 && asinfix.args[1] == asinfix.args[2] == :x    
        x = asinfix.head
    end   
end   
```
I'm not entirely sure if that breaks other things.